packages/payload/src/collections/config/types.ts

```diff
@@ -552,11 +552,21 @@ export type CollectionConfig<TSlug extends CollectionSlug = any> = {
   orderable?: boolean
   slug: string
   /**
-   * Add `createdAt` and `updatedAt` fields
+   * Add `createdAt`, `deletedAt` and `updatedAt` fields
    *
    * @default true
    */
   timestamps?: boolean
+  /**
+   * Enables trash support for this collection.
+   *
+   * When enabled, documents will include a `deletedAt` timestamp field.
+   * This allows documents to be marked as deleted without being permanently removed.
+   * The `deletedAt` field will be set to the current date and time when a document is trashed.
+   *
+   * @default false
```

Comment:
Let's add `@experimental` and call out that this is a beta feature since it is really new and touches so much.